Feat/create track

Anurag-Wednesday opened the pull request:
### Ticket Link

---

### Related Links

---

### Description

Added option to createTracks using Antd components
--user can now enter track details manually.

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---
![2022-02-14 16 38 48](https://user-images.githubusercontent.com/98798828/153859596-88423813-f0ef-4796-a114-2656fa898d8e.gif)

Anurag-Wednesday committed:
added option to upld details using antd components

Anurag-Wednesday committed:
added option to upld tracks using antd components

Anurag-Wednesday committed:
used translated strings

Anurag-Wednesday committed:
Merge branch 'feat/UAT' into feat/CreateTrack

Anurag-Wednesday committed:
fixed incorrect id

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,202 @@
+            <StyledInput data-testid="artist-name" />
+          </StyledFormItem>
+          <StyledFormItem
+            label={<FormattedMessage id="input-collection" />}
```
let's use T instead of FormattedMessage for consistency

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,202 @@
+          </StyledFormItem>
+          <StyledFormItem wrapperCol={{ offset: 8, span: 16 }}>
+            <Button type="primary" htmlType="submit" data-testid="submit">
+              Next
```
react-intl

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,202 @@
+
+  const contentOne = () => {
+    return (
+      <>
```
the Fragment seems unneccessary. We're wrapping it in a parent Form already right?

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,202 @@
+    setCurrent(current - 1);
+  };
+
+  const contentOne = () => {
```
all functions that are responsible for "rendering" should be prefixed with `render` 

Let's please also rename to depict what it is rendering rather than "contentOne"

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,202 @@
+
+  const contentTwo = () => {
+    return (
+      <>
```
same here about the name of the function and the fragment

Anurag-Wednesday commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,202 @@
+        <div className="steps-action">
+          {current > 0 && (
+            <Button data-testid="previous" style={{ margin: '0 8px' }} onClick={() => prev()}>
+              Previous
```
react-intl

Anurag-Wednesday commented on app/components/UploadTrack/tests/index.test.js:
```diff
@@ -0,0 +1,65 @@
+    const trackName = 'Sad';
+    const artistName = 'Sad';
+    const collection = 'Sad';
+    fireEvent.change(getByTestId('track-input'), { target: { value: trackName } });
```
```
const generator  = (value) => return ({ target: { value }} )
```

Anurag-Wednesday committed:
Made changes as per the comments

1. Used React-Intl wherever it was missed
2. Split the content for steps in new components
3. fixed casing issues with Id's
4. added tests to ensure complete coverage

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,82 @@
+import * as colors from '@app/themes/colors';
+import { TrackDetailsComponent } from '../TrackDetailsComponent/index';
+import UploadTrackBaseDetails from '../UploadTrackBaseDetails/index';
+import If from '../If/index';
```
please use @components instead of relative paths

alichherawalla replied:
also nothing wrong with `/index` but it reads a bit better without it. 

so `@components/TrackDetailsComponent` instead of `'../TrackDetailsComponent/index'`

alichherawalla replied:
we need to do this for all of the imports

alichherawalla replied:
If and TrackDetailsComponent too. 

Also as a practice we always import default exports instead of named imports for components. 


```
import { TrackDetailsComponent } from '../TrackDetailsComponent/index';
```
->
```
import TrackDetailsComponent from '@components/TrackDetailsComponent';
```

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,82 @@
+  );
+}
+
+UploadTrack.propTypes = {};
```
we can remove this bit if there are no props

alichherawalla commented on app/utils/trackCreationUtils.js:
```diff
@@ -0,0 +1,11 @@
+export const TRACK_CREATION_STEPS = [
+  {
+    title: 'Track Basic Details'
```
we should have the id here, and then translate in the presentational layer

Anurag-Wednesday committed:
fixed imports and used ids to translate step title

alichherawalla commented on app/components/UploadTrack/index.js:
```diff
@@ -0,0 +1,84 @@
+import UploadTrackImage from '@components/UploadTrackImage';
+import For from '@components/For';
+import { TRACK_CREATION_STEPS } from '@app/utils/TrackCreationUtils';
+import { T } from '@components/T';
```
import T from '@components/T';

alichherawalla commented on app/components/UploadTrackBaseDetails/index.js:
```diff
@@ -0,0 +1,74 @@
+import PropTypes from 'prop-types';
+import { Button, Form, Input } from 'antd';
+import { colors } from '@app/themes/index';
+import { T } from '../T/index';
```
here as well please

Anurag-Wednesday committed:
fixed issue for T import

Anurag-Wednesday committed:
updated import statements
